asapdiscovery-docking/asapdiscovery/docking/openeye.py

```diff
@@ -291,36 +307,71 @@ def _dock(
                         posed_mol = result.GetPose()
                         prob = result.GetProbability()
 
+                        # clear SD data for uniformity of results
+                        from asapdiscovery.data.backend.openeye import clear_SD_data
+
+                        clear_SD_data(posed_mol)
+
                         posed_ligand = Ligand.from_oemol(posed_mol, **set.ligand.dict())
                         # set SD tags
                         sd_data = {
                             DockingResultCols.DOCKING_CONFIDENCE_POSIT.value: prob,
                             DockingResultCols.POSIT_METHOD.value: oedocking.OEPositMethodGetName(
                                 result.GetPositMethod()
                             ),
+                            "Pose_ID": i,
                         }
                         posed_ligand.set_SD_data(sd_data)
+                        posed_ligands.append(posed_ligand)
 
                         # Generate info about which target was actually used by multi-reference docking
                         if isinstance(set, DockingInputMultiStructure):
                             docked_target = set.complexes[result.GetReceptorIndex()]
-                            input_pair = DockingInputPair(
-                                ligand=set.ligand, complex=docked_target
+                            input_pairs.append(
+                                DockingInputPair(
+                                    ligand=set.ligand, complex=docked_target
+                                )
                             )
                         else:
-                            input_pair = set
+                            input_pairs.append(set)
+
+                    # create hashable dict of input pairs
+                    input_pair_dict = {
+                        input_pair.unique_name: input_pair for input_pair in input_pairs
+                    }
+
+                    # split results by input pair
+                    from collections import defaultdict
 
+                    results_dict = defaultdict(list)
+                    for input_pair, posed_ligand in zip(input_pairs, posed_ligands):
+                        results_dict[input_pair.unique_name].append(posed_ligand)
+
+                    # return results split by input pair
+                    for input_pair_name, posed_ligands in results_dict.items():
                         docking_result = POSITDockingResults(
-                            input_pair=input_pair,
-                            posed_ligand=posed_ligand,
-                            probability=prob,
+                            input_pair=input_pair_dict[input_pair_name],
+                            posed_ligand=Ligand.from_single_conformers(posed_ligands),
                             provenance=self.provenance(),
+                            probability=(
+                                posed_ligands[0].tags[
```

Comment:
Can't this vary by ligand? Or are they all going to be homogenous?

Reply:
(sorry for the delay) what are you referring to by "this"?